misk/src/main/kotlin/misk/web/jetty/WebActionsServlet.kt

```diff
@@ -38,19 +43,48 @@ internal class WebActionsServlet @Inject constructor(
         keyOf<Request>() to asRequest)
 
     scope.enter(seedData).use {
-      val candidateActions = boundActions.mapNotNull { it.match(request, asRequest.url) }
+      val candidateActions = boundActions.mapNotNull {
+        it.match(
+            request.method,
+            request.contentType?.let { MediaType.parse(it) },
+            request.accepts(),
+            asRequest.url
+        )
+      }
       val bestAction = candidateActions.sorted().firstOrNull()
       bestAction?.handle(asRequest, response)
       logger.debug { "Request handled by WebActionServlet" }
     }
   }
+
+  override fun configure(factory: WebSocketServletFactory?) {
+    factory!!.creator = WebSocketCreator { servletUpgradeRequest, servletUpgradeResponse ->
+      val request = servletUpgradeRequest.httpServletRequest
+      val asRequest = Request(
```

Comment:
does request.asRequest() work?

Reply:
nit: change the signature to remove the nullability and get rid of the ``!!``